include/unicode.h

```diff
@@ -9,6 +9,17 @@
 
 #define UTF8_INVALID 0x80
 
+/**
+ * utf8_nr_bytes_to_prev_character - find distance to previous utf8 character
+ * @buf: C string buffer containing utf8 data
+ * @pos: position to start working backwards from
+ *
+ * This function does not validate that the buffer contains correct utf8 data;
+ * it merely looks for the first byte that correctly denotes the beginning of a
+ * utf8 character.
+ */
+int utf8_nr_bytes_to_prev_character(const char *buf, const char *pos);
```

Comment:
Maybe `utf8_last_size` would be a better name, for consistency with `utf8_size`?

I don't think we need to pass two arguments, a single should be enough. `pw->buffer` is zero-terminated.

Nit: it's very nice to add docs, but we don't use `@`-style argument descriptions. Plain English sentences to describe args can be used instead.

Reply:
Renamed to `utf8_last_size()`
Only pass `const char *str` and rely on it being `\0` terminated.
Local style used for documentation.
Have done `git commit --amend` and force-pushed. Hope that's okay.